.github/workflows/emscripten.yml

```diff
@@ -52,14 +52,23 @@ jobs:
           actions-cache-folder: emsdk-cache
 
       - name: Install pyodide-build
-        run: pip install "pydantic<2" pyodide-build==$PYODIDE_VERSION
+        run: pip install "pydantic<2" build pyodide-build==$PYODIDE_VERSION
 
       - name: Build
         run: |
-          # Pyodide is still in the process of adding better/easier support for
-          # non-setup.py based builds.
-          cp pyproject.toml.setuppy pyproject.toml
-          CFLAGS=-g2 LDFLAGS=-g2 pyodide build
+          # Note that we need to cross-compile here. Using `pyodide build`
+          # fails on not having a cross file. The `source() function in
```

Comment:
This is no longer true, the cross file is shipped by pyodide now.

Reply:
Update: still true actually, the cross file won't get applied as long as we have `vendored-meson`.

We also need the fix from https://github.com/pyodide/pyodide/pull/4502 to actually pick up `vendored-meson`. With that, an out of tree build should work.

Reply:
I am on it – I shall be adding a patch file to port the above PR into `vendored-meson`, as discussed on Slack. 

<!-- just wanted to post publicly -->